public/app/features/explore/RichHistory/RichHistoryCard.tsx

```diff
@@ -176,10 +178,13 @@ export function RichHistoryCard(props: Props) {
     appEvents.emit(AppEvents.alertSuccess, ['Query copied to clipboard']);
   };
 
-  const onCreateLink = () => {
-    const url = createUrlFromRichHistory(query);
-    copyStringToClipboard(url);
-    appEvents.emit(AppEvents.alertSuccess, ['Link copied to clipboard']);
+  const onCreateLink = async () => {
+    const link = createUrlFromRichHistory(query);
+    const shortLink = await createShortLink(link);
+    if (shortLink) {
```

Comment:
would it be worth to fire
```ts
      appEvents.emit(AppEvents.alertError, ['Error generating shortened link']);
```
so that we give feedback in case something goes wrong? (not sure on the wording and if `alertError` actually exists)

maybe this could even go inside `createShortLink` so that it would work also for the Explore Toolbar

what do you think?

Reply:
Good question! I had it there before, but then we have received 2 error messages in red rectangles. 1 from backend that bubbled up and 1 that I've created. And I just thought, that 1 error message might be enough.

Reply:
I agree we should show the error. Right now the createShortlink is memoized (preventing server request for the same urls) so showing alert there would not work. But I think we could wrap both success and error alert in a function as the code seem the same here and in explore.